src/Features/LanguageServer/Protocol/Handler/References/FindAllReferencesHandler.cs

```diff
@@ -41,6 +41,12 @@ public async Task<LSP.VSReferenceItem[]> HandleRequestAsync(ReferenceParams refe
                 return Array.Empty<LSP.VSReferenceItem>();
             }
 
+            // We only support streaming results back, so no point doing any work if the client doesn't
+            if (referenceParams.PartialResultToken == null)
```

Comment:
Once pull diagnostics goes in we should switch this to use the progress implementation there.
Filed https://github.com/dotnet/roslyn/issues/48350

Reply:
I already logged #48150 (and #48151) for this :P
I started working on, but need to make changes to the BufferedProgress to make it thread safe, so yes, was waiting for pull diagnostics to be in first.

Reply:
Woops, closed mine as dupe!